### ISSUE TITLE
Refactor: Signal calculation logic duplicated across 4 files — extract to signals.py

### ISSUE THREAD

**Issue by artfhc** [artfhc/clawdbot-spy-signals#17]
## Description

The core signal calculation (download data → compute MAs/RSI/Vol → evaluate 5 boolean conditions) is copy-pasted across four files with minor variations:

| File | Lines | Duplicated logic |
|------|-------|-----------------|
| `daily_signal_checker.py` | 14–53 | Full signal calc |
| `alpaca_trader.py` | 27–56 | Full signal calc |
| `signal_alerts.py` | 27–60 | Full signal calc |
| `dashboard.py` | 27–55 | Full signal calc (with extras) |

Any fix to the core logic (e.g., RSI division by zero, warmup check) must be applied in all four places. This is the root cause of issue #2 (Strategy E inconsistency) and makes issues #1, #5, #6 harder to fix atomically.

## Fix

Create `signals.py`:

```python
# signals.py
import yfinance as yf
import pandas as pd
import numpy as np
from config import RSI_OVERBOUGHT, MAX_VOLATILITY, MA_SLOPE_LOOKBACK

def fetch_and_compute(ticker: str, period: str = '2y') -> pd.DataFrame:
    """Download price data and compute all technical indicators."""
    data = yf.download(ticker, period=period, progress=False)
    if data.empty or len(data) < 250:
        raise ValueError(f"Insufficient data for {ticker}")
    if isinstance(data.columns, pd.MultiIndex):
        data.columns = data.columns.get_level_values(0)
    # ... compute MA20, MA50, MA200, MA200_Slope, RSI, Vol ...
    return data

def get_signals(data: pd.DataFrame) -> dict:
    """Return dict of signal booleans for the latest row."""
    latest = data.iloc[-1]
    loss = ...  # safe RSI calc
    ...
    return {'trend': ..., 'golden': ..., 'momentum': ..., 'rsi_ok': ..., 'low_vol': ...}
```

Then each consuming file does:
```python
from signals import fetch_and_compute, get_signals
```
